meshparty/trimesh_io.py

```diff
@@ -405,7 +407,10 @@ def n_faces(self):
 
     @property
     def mesh_edges(self):
-        return self._mesh_edges
+        if self._non_face_edges is None:
+            return self.edges
+        else:
+            return np.vstack([self.edges, self._non_face_edges])
```

Comment:
this is not the agreed upon behavior. vstack is executed every single time this property is called... 
This would need t be designed using an lru cache or something similar.

Reply:
I have now integrated the trimesh caching system, this simplified lots of code, including many of our other properties.